pkg/lib/template/types.go

```diff
@@ -0,0 +1,8 @@
+package template
+
+type Parser interface {
+	// Parse parses the template and replaces the placeholders with the values a replacements map.
+	Parse(content string) (string, error)
+	// ParseBytes parses the template and replaces the placeholders with the values a replacements map.
+	ParseBytes(content []byte) ([]byte, error)
+}
```

Comment:
Is this actually used anywhere? Could probably be removed.

Reply:
it is the interface that `DefaultParser` is implementing. Will add a compile time check